### DISCUSSION TITLE
How to complete the given task.

### DISCUSSION THREAD

**Discussion by Sapnajha19** [open-sauced-craftwork/browser-extension#42] [Q&A]
I successfully set up the project and created an extension by following [CRXJS](https://crxjs.dev/vite-plugin/getting-started/react/create-project), but now I don't understand how to proceed further. Am I supposed to create the "View on opensauced" button in the extension?

**Comment by mukulpadwal** [comment]
@Sapnajha19 The button should be visible on the GitHub profile page of a particular user if he/she has an account on opensauced else do nothing.

**Comment by Sapnajha19** [comment]
I completed the button's functionality but even when  different users click the button(who have signed up on open-sourced) it still opens my page on opensauced not the user's page.
 And also my opensauced profile is not showing my profile like other people. Its showing insight instead of profile.
![Insights Hub _ Open Sauced Insights - Google Chrome 24-03-2023 23_30_40](https://user-images.githubusercontent.com/99778524/227604556-b7c850dd-0c99-4261-bb45-8b38848d19eb.png)

**Reply by mukulpadwal** [reply]
@Sapnajha19 i guess you might have hard coded the value of GitHub username inside your code.


So you can do something like this 

When you open a particular users GitHub page then fetch the username and accordingly make the api call to check if the user is there on opensauced or not.


The response which you will get you can use that to render the button.


If(user has a opensauced account)
     Display the button

Else leave do nothing.

**Reply by Rishav1707** [reply]
Use `document.location.pathname.split('/')[1]` to fetch the username from the GitHub profile.
And in the content scripts set the matches to GitHub only (by default it uses google.com).
